src/tests/class_tests/openms/source/TargetedSpectraExtractor_test.cpp

```diff
@@ -0,0 +1,837 @@
+  // {
+  //   return a.getFloatDataArrays()[1][0] > b.getFloatDataArrays()[1][0];
+  // });
+  // sort(scored_spectra.begin(), scored_spectra.end(), [](MSSpectrum a, MSSpectrum b)
+  // {
+  //   return a.getName().compare(b.getName()) < 0;
+  // });
+  // STATUS("Scored spectra have been sorted by name.")
+
+  STATUS("Info from scored spectra:")
+  for (auto s : scored_spectra)
+  {
+    STATUS(s.getName()
+    << "\t score: " << s.getFloatDataArrays()[1][0]
+    << "\t log10_tic: " << s.getFloatDataArrays()[2][0]
+    << "\t 1/fwhm: " << s.getFloatDataArrays()[3][0]
+    << "\t SNR: " << s.getFloatDataArrays()[4][0])
+  }
+
+  STATUS("Info from FeatureMap:")
+  for (auto f : features)
+  {
+    STATUS(f.getMetaValue("transition_name")
+    << "\t score: " << f.getIntensity()
+    << "\t log10_tic: " << f.getMetaValue("log10_total_tic")
+    << "\t 1/fwhm: " << f.getMetaValue("inverse_avgFWHM")
+    << "\t SNR: " << f.getMetaValue("avgSNR")
+    << "\t fwhm: " << f.getMetaValue("avgFWHM"))
+  }
+}
+END_SECTION
+
+START_SECTION(selectSpectra())
+{
+  ptr->setUseGauss(true);
+  ptr->setGaussWidth(0.25);
+  ptr->setRTWindow(30.0);
+  ptr->setMZTolerance(0.1);
+  ptr->setPeakHeightMin(15000.0);
+  ptr->setPeakHeightMax(110000.0);
+  ptr->setFWHMThreshold(0.23);
+  ptr->setTICWeight(1.0);
+  ptr->setFWHMWeight(1.0);
+  ptr->setSNRWeight(1.0);
+  ptr->setMinScore(15.0);
+
+  const std::vector<MSSpectrum> spectra = experiment.getSpectra();
+  std::vector<MSSpectrum> annotated;
+  FeatureMap features;
+  ptr->annotateSpectra(spectra, targeted_exp, annotated, features);
+  std::vector<MSSpectrum> picked(annotated.size());
+  for (UInt i=0; i<annotated.size(); ++i)
+  {
+    ptr->pickSpectrum(annotated[i], picked[i]);
+  }
+  for (Int i=annotated.size()-1; i>=0; --i)
+  {
+    if (!picked[i].size())
+    {
+      annotated.erase(annotated.begin() + i);
+      picked.erase(picked.begin() + i);
+      features.erase(features.begin() + i);
+    }
+  }
+  std::vector<MSSpectrum> scored;
+  ptr->scoreSpectra(annotated, picked, features, scored);
+
+  STATUS("Scored spectra and their score:")
+  for (auto s : scored)
+  {
+    STATUS(s.getName() << "\t" << s.getFloatDataArrays()[1][0])
+  }
+
+  std::vector<MSSpectrum> selected_spectra;
+  FeatureMap selected_features;
+
+  ptr->selectSpectra(scored, features, selected_spectra, selected_features);
+  TEST_EQUAL(selected_spectra.size(), 2)
+  TEST_EQUAL(selected_spectra.size(), selected_features.size())
+  for (UInt i=0; i<selected_spectra.size(); ++i)
+  {
+    TEST_NOT_EQUAL(selected_spectra[i].getName(), "")
+    TEST_EQUAL(selected_spectra[i].getName(), selected_features[i].getMetaValue("transition_name"))
+    TEST_EQUAL(selected_spectra[i].getFloatDataArrays()[1][0], selected_features[i].getIntensity())
+    TEST_EQUAL(selected_spectra[i].getFloatDataArrays()[1][0] >= ptr->getMinScore(), true)
+  }
+
+  ptr->selectSpectra(scored, selected_spectra);
+  TEST_NOT_EQUAL(selected_spectra.size(), 0)
```

Comment:
can you please test for something meaningful, e.g. the actual number of selected spectra?